gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/workunit/packer/KafkaTopicGroupingWorkUnitPacker.java

```diff
@@ -381,4 +391,40 @@ static double getContainerCapacityForTopic(List<Double> capacities, ContainerCap
         throw new RuntimeException("Unsupported computation strategy: " + strategy.name());
     }
   }
+
+  /**
+   * A method that split a list of {@link MultiWorkUnit} to the size of desiredWUSize if possible. The split approach is to
```

Comment:
Add comments here to mention that we always try to split MWU with more partitions first?